router/oauthResponseHandler/oauthResponseHandler.go

```diff
@@ -0,0 +1,178 @@
+package oauthResponseHandler
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	backendconfig "github.com/rudderlabs/rudder-server/config/backend-config"
+	"github.com/rudderlabs/rudder-server/services/stats"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+)
+
+type AccountSecret struct {
+	AccessToken    string `json:"accessToken"`
+	ExpirationDate string `json:"expirationDate"`
+}
+
+// OAuthErrResHandler is the handle for this class
+type OAuthErrResHandler struct {
+	tr                             *http.Transport
+	client                         *http.Client
+	oauthErrHandlerReqTimerStat    stats.RudderStats
+	oauthErrHandlerNetReqTimerStat stats.RudderStats
+	logger                         logger.LoggerI
+}
+
+type Authorizer interface {
+	Setup()
+	DisableDestination(destination backendconfig.DestinationT, workspaceId string) (statusCode int, resBody string)
+	RefreshToken(workspaceId string, accountId string, accessToken string) (statusCode int, resBody string)
+}
+
+// This function creates a new OauthErrorResponseHandler
+func NewOAuthErrorHandler() *OAuthErrResHandler {
+	return &OAuthErrResHandler{}
+}
+
+var (
+	configBEURL string
+	pkgLogger   logger.LoggerI
+	loggerNm    string
+)
+
+const (
+	DISABLE_DEST  = "DISABLE_DESTINATION"
+	REFRESH_TOKEN = "REFRESH_TOKEN"
+)
+
+// The response from the transformer network layer will be sent with a output property(in case of an error)
+type ErrorOutput struct {
+	Output ErrorResponse `json:"output"`
+}
+
+// This struct represents the datastructure present in Transformer network layer Error builder
+type ErrorResponse struct {
+	Message           string                 `json:"message"`
+	Destination       map[string]interface{} `json:"destination"`
+	NetworkFailure    bool                   `json:"networkFailure"`
+	Status            int                    `json:"status"`
+	AuthErrorCategory string                 `json:"authErrorCategory"`
+	AccessToken       string                 `json:"accessToken"`
+}
+
+type RefreshTokenBody struct {
+	hasExpired   bool
+	expiredToken string
+}
+
+// Custom Marshalling for RefreshTokenBody struct
+func (refTokenBody RefreshTokenBody) MarshalJSON() ([]byte, error) {
+	j, err := json.Marshal(struct {
+		HasExpired   bool
+		ExpiredToken string
+	}{
+		HasExpired:   refTokenBody.hasExpired,
+		ExpiredToken: refTokenBody.expiredToken,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return j, nil
+}
+
+func init() {
```

Comment:
https://github.com/rudderlabs/rudder-server/pull/1176/files

We have depricated the use of init(). Please refer the above PR